rs/registry/node_provider_rewards/src/lib.rs

```diff
@@ -8,6 +8,7 @@ use std::collections::{BTreeMap, HashMap};
 
 pub struct RewardsPerNodeProvider {
     pub rewards_per_node_provider: BTreeMap<PrincipalId, u64>,
+    pub computation_log: BTreeMap<PrincipalId, Vec<String>>,
```

Comment:
My main concern is that we are creating an API here, where the consumers will have to parse strings if they want to do anything beyond just reading it (i.e. they want to write a program to analyze the logs or do some automatic checking).

I'm wondering if we could use enums to represent each log line so that it represents the same information (i.e. rate not found in reward table) but would be a more concise representation?  We could then have a Display impl for each thing, so we could still print it out just as easily.

WDYT?

Reply:
Yeah it sounds reasonable to me, we would need it anyway also with the "new rewarding algo" to display the log. Let me come up with something ;)